### ISSUE TITLE
There's no way to stop a server other than exiting the program

### ISSUE THREAD

**Issue by Trafalg** [icza/gowut#14]
This seems like a problem if you want to, say, include a feature to change the port via the gowut interface, but can't actually shut down and restart the server.

Googling indicates that it is possible. See for example https://github.com/tylerb/graceful

**Comment by willauld** [comment]
I'd second this enhancement. Would the following be directly useful?
`https://golang.org/pkg/net/http/#Server.Shutdown`

**Comment by icza** [comment]
Yes, the [graceful shutdown addition in Go 1.8](https://golang.org/doc/go1.8#http_shutdown) can be used for this.
